### ISSUE TITLE
Getting error while trying to build the project

### ISSUE THREAD

**Issue by kshwetabh** [josh-collinsworth/sveltekit-blog-starter#20]
Hello, I am pretty new to Svelte world and landed here following your awesome [post](https://joshcollinsworth.com/blog/build-static-sveltekit-markdown-blog) on how to create a MD blog using sveltekit. Thank you for all the effort, I played around with the project in development mode and the level of time and effort put into this amazing work shows well.

When I try to build the project (after cloning it from github), it errors out with the following message. Any idea what I am doing wrong ?
Error message gives no clue as to what's missing.

**Steps executed:**

1. Clone the repo
> git clone https://github.com/josh-collinsworth/sveltekit-blog-starter.git
2. CD into sveltekit-blog-starter and install the dependencies.
> cd sveltekit-blog-starter
> npm install
3. Run the project. Everything works fine am able to open every single post and page.
> npm run dev
5. Build the project:
> npm run build
6. Throws the below error.

``` js
. . .
.svelte-kit/output/server/chunks/README.js                                                     11.28 KiB
  301 /blog/page -> /blog
  301 /blog/category/page/* -> /blog
  301 /blog/page/* -> /blog
node:fs:585
  handleErrorFromBinding(ctx);
  ^

Error: ENOENT: no such file or directory, open 'C:\Dev\nodejs\svelte-static-blog\sveltekit-blog-starter\.svelte-kit/output/prerendered/pages/api/posts/category/*'
    at Object.openSync (node:fs:585:3)
    at writeFileSync (node:fs:2155:35)
    at save (file:///C:/Dev/nodejs/svelte-static-blog/sveltekit-blog-starter/node_modules/@sveltejs/kit/src/core/prerender/prerender.js:317:4)
    at visit (file:///C:/Dev/nodejs/svelte-static-blog/sveltekit-blog-starter/node_modules/@sveltejs/kit/src/core/prerender/prerender.js:196:3) {
  errno: -4058,
  syscall: 'open',
  code: 'ENOENT',
  path: 'C:\\Dev\\nodejs\\svelte-static-blog\\sveltekit-blog-starter\\.svelte-kit/output/prerendered/pages/api/posts/category/*'
}
[vite-plugin-svelte-kit] Prerendering failed with code 1
error during build:
Error: Prerendering failed with code 1
    at ChildProcess.<anonymous> (file:///C:/Dev/nodejs/svelte-static-blog/sveltekit-blog-starter/node_modules/@sveltejs/kit/src/exports/vite/index.js:443:15)
    at ChildProcess.emit (node:events:520:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
```

If it helps, this is the content of my package.json file:

```json
{
  "name": "sveltekit-blog-starter",
  "version": "1.4.2",
  "scripts": {
    "dev": "vite dev",
    "build": "vite build",
    "preview": "vite preview",
    "lint": "prettier --ignore-path .gitignore --check --plugin-search-dir=. . && eslint --ignore-path .gitignore .",
    "format": "prettier --ignore-path .gitignore --write --plugin-search-dir=. ."
  },
  "devDependencies": {
    "@sveltejs/adapter-static": "^1.0.0-next.42",
    "@sveltejs/kit": "next",
    "eslint": "^7.32.0",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-svelte3": "^3.2.1",
    "mdsvex": "^0.9.8",
    "prettier": "^2.4.1",
    "prettier-plugin-svelte": "^2.4.0",
    "rehype-autolink-headings": "^6.1.0",
    "rehype-slug": "^5.0.0",
    "sass": "^1.43.4",
    "svelte": "^3.42.6",
    "vite": "^3.1.0"
  },
  "type": "module",
  "dependencies": {
    "just-throttle": "^4.0.1",
    "svelte-preprocess": "^4.9.8"
  }
}
```

**Comment by josh-collinsworth** [comment]
Thanks for the detailed report @kshwetabh! Hmmmm odd, I went through the same steps and didn't run into that issue. Everything ran and built fine for me.

However, I _did_ run into a separate issue when attempting to update the site. So I suspect perhaps the issue has to do with the `next` version of SvelteKit. I'll keep this open for the time being, since it seems like a patch will likely be forthcoming quickly. If this isn't resolved for you soon, let me know.

The only other thing I can think is that it's an issue with either the installed version of Node.

**Comment by josh-collinsworth** [comment]
Circling back to this: I just updated some of the packages. It may be worth re-trying and seeing if you get a better result now.

**Comment by kshwetabh** [comment]
Thanks for your time @josh-collinsworth. Still no luck. Giving the same error.  I am running Node v16.18.1.
I noticed that during the build process the output category directory is empty possibly causing the error/build failure.

![image](https://user-images.githubusercontent.com/970301/204797664-837f88b6-7870-485f-8d2b-a29fefbed024.png)

**Comment by josh-collinsworth** [comment]
@kshwetabh Ok, I'm not sure why it's causing an error on your end and not mine (not even if I change my Node version to match, weirdly), but I _do_ think I've figured out what the root issue might be.

I've realized the `/api/posts/category/*` routes aren't prerendering properly because they're _never actually used_ anywhere on the site; when SvelteKit's prerender crawler doesn't see a link to something, it doesn't know what to generate, and so we just end up with the `*` folder.

Since the category routes in the API are never used anywhere anyway, I just pushed an update that removes those (both from `src/routes`, and from the config's `kit.prerender.entries` list).

Try a fresh install now, and let me know if you get a better/different result.

EDIT: I have a theory about why it might error on your end and not mine; `*` is a forbidden folder name on Windows, but not on MacOS/Linux/Unix. So maybe the build errors on Windows when it generates a `*` directory, in a way that it doesn't on Mac.
